snuba/datasets/table_storage.py

```diff
@@ -155,24 +162,35 @@ def get_bulk_writer(self, options=None, table_name=None) -> BatchWriter:
         # once we will be confident it is reliable enough.
 
         from snuba import settings
-        from snuba.clickhouse.http import HTTPBatchWriter
 
-        return HTTPBatchWriter(
-            self.__table_schema,
-            settings.CLICKHOUSE_HOST,
-            settings.CLICKHOUSE_HTTP_PORT,
+        table_name = table_name or self.__table_schema.get_table_name()
+
+        options = self.__update_writer_options(options)
+
+        return self.__cluster.get_writer(
+            table_name,
             lambda row: rapidjson.dumps(row).encode("utf-8"),
             options,
-            table_name,
-            chunk_size=settings.BULK_CLICKHOUSE_BUFFER,
+            chunk_size=settings.CLICKHOUSE_HTTP_CHUNK_SIZE,
         )
 
-    def get_bulk_loader(self, source, dest_table) -> BulkLoader:
+    def get_bulk_loader(
```

Comment:
I'll keep this method here for consistency even if it doesn't provide much. Ideally I think this would move into the CdcStorage class though, and then only need to take source and dest_table as arguments.

Reply:
On second thoughts.... will pass the connection in here instead of in the load method.